hobbit-gui/gui-serverbackend/src/main/java/de/usu/research/hobbit/gui/rest/ExperimentsResources.java

```diff
@@ -65,28 +67,81 @@
         if (Application.isUsingDevDb()) {
             return getDevDb().queryExperiments(ids, challengeTaskId);
         } else {
+            // get user
+            UserInfoBean userInfo = InternalResources.getUserInfoBean(sc);
+            List<SystemBean> userSystems = PlatformControllerClientSingleton.getInstance().requestSystemsOfUser(userInfo.getPreferredUsername());
+            // create set of user owned system ids
+            String[] sysIds = userSystems.stream().map(s -> s.getId()).toArray(String[]::new);
+            Set<String> userOwnedSystemIds = new HashSet<>(Arrays.asList(sysIds));
+
             if (ids != null) {
                 System.out.println("Querying experiment results for " + Arrays.toString(ids));
                 results = new ArrayList<>(ids.length);
                 for (String id : ids) {
                     // create experiment URI
                     String experimentUri = Constants.EXPERIMENT_URI_NS + id;
-                    String query = SparqlQueries.getExperimentGraphQuery(experimentUri, null);
-                    // TODO make sure that the user is allowed to see the
-                    // experiment!
+                    // construct public query
+                    String query = SparqlQueries.getExperimentGraphQuery(experimentUri, Constants.PUBLIC_RESULT_GRAPH_URI);
+                    // get public experiment
                     Model model = StorageServiceClientSingleton.getInstance().sendConstructQuery(query);
                     if (model != null) {
```

Comment:
I think that it is possible that there is a model returned which is empty so it should be checked whether there are any triples in the model.